EUDAT-B2SAFE/B2SAFE-core
# Add integration tests for EPIC client

This adds integration tests for the [EPICv2 client library](https://github.com/EUDAT-B2SAFE/B2SAFE-core/blob/master/cmd/epicclient.py). The suite of tests includes the test cases previously implemented through the EPIC API's `test` method (removed as of 210e6ff) and adds new cases that increase code coverage. These tests cases are on par with the [EPIC API unit tests](https://github.com/EUDAT-B2SAFE/B2SAFE-core/blob/master/scripts/tests/testB2SafeCmd/epicapitest.py) but, instead of mocking REST calls, they invoke an actual EPIC service instance (and are thus slower).

#### How to run the tests ####
The EPIC client integration tests can be run together with the unit tests (see also #28):
```
[B2SAFE-core/scripts/tests]% python -m coverage run testB2SafeCmd.py -test epic && python -m coverage xml
```
This will run all unit and integration EPIC client test suites and create an xml-formatted coverage report suitable for Jenkins/SonarQube

**N.B.** The integration tests will be skipped unless a valid credentials file named `epic_credentials` is available under the [`resources`](https://github.com/EUDAT-B2SAFE/B2SAFE-core/tree/master/scripts/tests/resources) directory.

/cc @TobiasWeigel
